### ISSUE TITLE
Don't apply syntax highlighting on addContext strings

### ISSUE THREAD

**Issue by irvinlim** [adamgruber/mochawesome-report-generator#37]
This is a follow-up from my previous issue, #35, where I briefly covered the issue with syntax highlighting on contexts added using `addContext`. 

For example, this is what I am passing in:

```js
addContext(this, {
    title: "HTTP Response",
    value: "HTTP/1.1 403 Forbidden\n...."
});
```

The following is an example of how syntax highlighting is wrongly applied:

<img width="285" alt="screen shot 2017-06-28 at 1 31 32 am" src="https://user-images.githubusercontent.com/9884746/27601334-971fc7dc-5ba1-11e7-8de1-98e93120f1ea.png">

This is what I would expect:

<img width="285" alt="screen shot 2017-06-28 at 1 31 24 am" src="https://user-images.githubusercontent.com/9884746/27601335-9726c6ea-5ba1-11e7-8fe8-5ad84e0f6140.png">

---

To summarize, there are currently 4 types of contexts that can be added:

1. Simple string, i.e. `addContext(this, 'this is a string')`, which will not apply syntax highlighting
2. Image URL, i.e. `addContext(this, 'https://images.com/cheezburger.gif')`
3. URL, i.e. `addContext(this, https://google.com)`
4. Object with the following schema:

```js
addContext(this, {
    title: "My Title",
    value: "" // anything here
})
```

which will render a context with a `h4` header, as shown above.

---

The problem arises when I use the 4th option (using an object) with a string, which causes the string to have JS highlighting when it is not supposed to.

A simple fix would be to turn off syntax highlighting for all string-typed `value` values by default, until we offer a third "options" argument for `addContext` (which we also discussed previously).

In other words, the following is my suggested implementation for `renderContextContent` in [`components/test/context.jsx`](https://github.com/adamgruber/mochawesome-report-generator/blob/d01cc795eb08756e6cf1067cffe6092552d2fa6a/src/components/test/context.jsx#L55):

```jsx
renderContextContent = (content, title) => {
  // Images
  if (imgRegEx.test(content)) {
    return this.renderImage(content, title);
  }

  // URLs
  if (urlRegEx.test(content)) {
    return this.renderLink(content, title);
  }

  // Simple strings
  if (isString(content)) {
    return <CodeSnippet className={ cx('code-snippet') } code={ content } highlight={ false } />;
  }

  // All other types, including primitives, objects and arrays should have syntax highlighting
  const code = JSON.stringify(content, null, 2);
  return (
    <CodeSnippet className={ cx('code-snippet') } code={ code } highlight />
  );
}
```

This might make sense because most people might be using this method to be passing in either strings, primitives, objects or arrays. For all types except for strings, applying syntax highlighting is fine and even encouraged, but as for strings, I'm not sure whether users will use it to pass in JavaScript code.

Do correct me if I'm wrong about that. Otherwise, I could create a PR if you are agreeable with this.

**Comment by adamgruber** [comment]
@irvinlim I was working on some other additions so I added this as well. Thanks for the write up!

**Comment by adamgruber** [comment]
Published v2.2.0 with this change
